app/services/record_declaration.rb

```diff
@@ -258,4 +260,14 @@ def create_participant_outcome!
       raise Api::Errors::InvalidParticipantOutcomeError, I18n.t(:cannot_create_completed_declaration)
     end
   end
+
+  def set_mentor_completion
+    Mentors::CompleteTraining.call(mentor_profile: participant_profile) if mentor_completion_event?
+  end
+
+  def mentor_completion_event?
+    participant_profile.mentor? &&
```

Comment:
I think we might need more validations in place here, especially ones mentioned around preventing providers from making declarations dated after a mentor has completed training. Happy to discuss

Reply:
Did we resolve this one? I have a feeling we didn't think this was an issue for this?